File: support_tools/swigwin-3.0.12/Examples/test-suite/python/smart_pointer_multi_runme.py
from smart_pointer_multi import *

f = Foo()
b = Bar(f)
s = Spam(b)
g = Grok(b)

s.x = 3
if s.getx() != 3:
    raise RuntimeError

g.x = 4
if g.getx() != 4:
    raise RuntimeError

File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/smart_pointer_multi_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#
# 
# 
# 
#

require 'swig_assert'

require 'smart_pointer_multi'

include Smart_pointer_multi

f = Foo.new
b = Bar.new(f)
s = Spam.new(b)
g = Grok.new(b)

s.x = 3
raise RuntimeError if s.getx() != 3

g.x = 4
raise RuntimeError if g.getx() != 4

